stock_picking_partner_vat/views/stock_picking_views.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_stock_picking_form_inherit" model="ir.ui.view">
+        <field name="name">stock.picking (vat)</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form"></field>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='partner_id']" position="after">
+                <field name="partner_vat" attrs="{'invisible': [('picking_type_code', 'not in', ('incoming', 'outgoing'))]}"/>
```

Comment:
I've realized that even an internal transfer could have a partner_id (but optional).
Then, maybe a simpler invisible condition should be this one:
```suggestion
                <field name="partner_vat" attrs="{'invisible': [('partner_vat', '=', False)]}"/>
```
What do you think?